src/ripple/app/ledger/impl/InboundLedgers.cpp

```diff
@@ -156,11 +157,26 @@ class InboundLedgersImp : public InboundLedgers
         InboundLedger::Reason reason) override
     {
         std::unique_lock lock(acquiresMutex_);
-        if (pendingAcquires_.contains(hash))
-            return;
-        pendingAcquires_.insert(hash);
-        lock.unlock();
-        acquire(hash, seq, reason);
+        try
+        {
+            if (pendingAcquires_.contains(hash))
+                return;
+            pendingAcquires_.insert(hash);
+            lock.unlock();
+            acquire(hash, seq, reason);
+        }
+        catch (std::exception const& e)
+        {
+            JLOG(j_.warn())
+                << "Exception thrown for acquiring new inbound ledger " << hash
+                << ": " << e.what();
+        }
+        catch (...)
+        {
+            JLOG(j_.warn())
+                << "Unknown exception thrown for acquiring new inbound ledger "
+                << hash;
+        }
         lock.lock();
         pendingAcquires_.erase(hash);
```

Comment:
The manual calls to `unlock` and `lock` still being here are rubbing me the wrong way. What about:
```suggestion
        try
        {
            {
                std::unique_lock lock(acquiresMutex_);
                if (pendingAcquires_.contains(hash))
                    return;
                pendingAcquires_.insert(hash);
            }
            acquire(hash, seq, reason);
        }
        catch (std::exception const& e)
        {
            JLOG(j_.warn())
                << "Exception thrown for acquiring new inbound ledger " << hash
                << ": " << e.what();
        }
        catch (...)
        {
            JLOG(j_.warn())
                << "Unknown exception thrown for acquiring new inbound ledger "
                << hash;
        }
        std::unique_lock lock(acquiresMutex_);
        pendingAcquires_.erase(hash);
```

Reply:
Sure, go ahead. Can also be lock_guard in that case if you want.

Reply:
I'm fine with leaving this as-is for now, but check out https://github.com/ximinez/rippled/commit/7a247645899edefdccf93eae404a79aacd68eb95